### PR TITLE
Plus one and flagging backend

### PR DESCRIPTION
**Pull request by labseven**
It's all there. Some of it could be fleshed out, and like everything else, needs more robust error checking.

### CODE REVIEW

File: models/bob.js [hunk 1]
```diff
@@ -3,11 +3,13 @@ const mongoose = require('mongoose');
...
-  data: {},
+  data:      {},
   startDate: { type: Date, default: Date.now() },
-  endDate: Date,
-  flavor: String,
-  tags: []
+  endDate:   Date,
+  flavor:    String,
+  tags:      [],
+  votes:     { type: Number, default: 1 },
```
Comment (SeunginLyu) [inline models/bob.js:11]: shouldn't default be zero?

File: routes/api.js [hunk 2]
```diff
@@ -108,6 +114,59 @@ module.exports = function(io, db) {
...
+    }, function error(err) {
+      res.status(500).send(err);
+    });
+  }
+
+  function POSTflagBob(req, res) {
+    console.log(req);
+    db.Bob.flagBob(db.ObjectId(req.params.bobid)).then(function success(data) {
+      if(data){
+        io.emit('delete', req.params.bobid);
```
Comment (SeunginLyu) [inline routes/api.js:158]: maybe naming the socket 'flag' is better?

File: models/bob.js [hunk 3]
```diff
@@ -62,6 +64,21 @@ function deleteBob(bobId) {
...
+  return BobModel.findOneAndUpdate(
+    { _id: bobId },
+    {
+      $inc: { votes: 1}  // Increment votes by 1
+    }
+  );
+}
+
+function flagBob(bobId) {
+  // console.log('flagging bob id: ' + bobId);
```
Comment (aidankmcl) [inline models/bob.js]: remove console log

**Review by aidankmcl — CHANGES_REQUESTED** [review]

File: models/bob.js (hunk 1 above)
Comment (labseven) [reply models/bob.js:11]: I like it start at 1 because then it defaults to +1 on the front page. +0 is a little sad.

**Pull request merged**